llama-index-integrations/vector_stores/llama-index-vector-stores-weaviate/llama_index/vector_stores/weaviate/utils.py

```diff
@@ -47,13 +47,15 @@ def validate_client(client: Any) -> None:
     """Validate client and import weaviate library."""
     try:
         import weaviate
-
-        client = cast(weaviate.WeaviateClient, client)
     except ImportError:
         raise ImportError(
             "Weaviate is not installed. "
             "Please install it with `pip install weaviate-client`."
         )
+    if not isinstance(client, weaviate.WeaviateClient):
+        raise ValueError(
+            f"Invalid client type, expected weaviate.WeaviateClient, got {type(client)}"
+        )
```

Comment:
This PR is slightly out of date -- you can technically pass in an async client or sync client these days. This would break that